src/scripts/ui/ResourcePanel.tsx

```diff
@@ -93,6 +104,7 @@ export function ResourcePanel(): React.ReactNode {
             aria-label={t(L.NotProducingBuildings)}
             data-balloon-pos="down"
             data-balloon-text="left"
+            onClick={() => cycleNotProducingReasons()}
```

Comment:
I like this idea but the implementation can be problematic (since the buildings that are not producing can change every cycle).
I've just commit the support to highlight multiple tiles, I think it's better (and cleaner) to use that API. When a player click the button, simply highlight all not producing buildings
```typescript
Singleton().sceneManager.getCurrent(WorldScene)?.drawSelection(null, tilesToHighlight);
```
You can refer to how this is implemented: https://github.com/fishpondstudio/CivIdle/blob/0a39ce9f6c8e062020376e42a9537ea7433924ad/src/scripts/ui/BuildingUpgradeComponent.tsx#L81

Reply:
Happy to change it, but I'm not seeing a problem, every time you click it gets a fresh list of the buildings, I could add a check if notProducingIdx is greater than the list length and reset it to 0.

Reply:
The potential issue is that if there's a new key inserted between cycles, then the `index + 1` *might* point to the same key.

Reply:
Yes I see the issue.  I have updated it to the tile highlight method.